twitchpopups.htm

```diff
@@ -102,24 +112,66 @@ <h1 id="popuptext"></h1>
                     $("#popuptext").text(commandName.substr(7).toUpperCase());
                     doAnimation();
                 } else if (commandName.startsWith("!delete")) {
-                  deleteAnimation();
+                    deleteAnimation();
                 } else if (commandName.startsWith("!hotseat ")) {
-                  hotSeatIsOn = true;
-                  $("#popupbox").show();
-                  hotSeatUser = commandName.substr(10).toLowerCase();
-                  $("#popupbox").css({"background-color":hotseatBg}); 
-                  $("#popuptext").text(`${hotseatEmoji}  ${hotSeatUser.toUpperCase()} IS IN THE HOTSEAT ${hotseatEmoji}`);
-                  doAnimation();
+                    hotSeatIsOn = true;
+                    $("#popupbox").show();
+                    voteTopic = commandName.substr(10);
+                    $("#popupbox").css({"background-color":hotseatBg}); 
+                    $("#popuptext").text(`${hotseatEmoji}  ${hotSeatUser.toUpperCase()}`);
+                    doAnimation();
+                } else if (commandName.startsWith("!vote ")) {
+                    // Start with the topic for your vote
+                    $("#popupbox").show();
+                    voteTopic = commandName.substr(6).toUpperCase();
+                    $("#popupbox").css({"background-color": votingBg});
+                    $("#popuptext").text(`VOTE: ${voteTopic} 🔴`);
+                    doAnimation();
+                } else if (commandName.startsWith("!voteopt ")) {
+                    // Then add your options
+                    let option = commandName.substr(9).toUpperCase();
+                    options[option] = {};
+                } else if (commandName.startsWith("!voteoptclear")) {
+                    // Made a mistake? clear and re-add your options
+                    options = [];

```

Comment:
Should this be reset to an object?

Reply:
Nope, it is an array but good spot on the initialisation, I have set that to be the correct structure.